### PR TITLE
feat: TE7-T117 Active Record Migrations

### PR DESCRIPTION
**Pull request by Dhyey54**

### CODE REVIEW

File: app/views/authors/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<h1 class="text-center m-4 h1">New Author</h1>
+
+<%= render "form", author: @author %>
```
Comment (poonamsimform) [inline app/views/authors/new.html.erb]: Add extra line

File: app/views/authors/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<div class="container w-50">
+<%= form_with model: author do |form| %>
```
Comment (poonamsimform) [inline app/views/authors/_form.html.erb]: Check form indentation.

File: app/views/authors/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<h1 class="text-center m-4 h1">Authors</h1>
+<table class="table w-50 table-bordered table-striped table-hover m-auto text-center">
+<tr>
+    <th>Id</th>
+    <th>Name</th>
+    <th>Date Of Birth</th>
+    <th>Email</th>
+  </tr>
```
Comment (poonamsimform) [inline app/views/authors/index.html.erb]: Check indentation

File: app/views/authors/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    <td><%= @author.id %></td>
+    <td><%= @author.first_name + " " + @author.last_name %></td>
+    <td><%= @author.date_of_birth %></td>
+    <td><%= @author.email %></td>
+</tr>
+
+</table>
+
+<h3 class="text-center m-4"> <%= @author.first_name + " " + @author.last_name %> has <%= @books.count %> books.</h3>
+
```
Comment (poonamsimform) [inline app/views/authors/show.html.erb]: Indentation

File: app/views/authors/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    <td><%= book.name %></td>
+    <td><%= book.price %></td>
+    <td><%= book.author_id %></td>
+  </tr>
+<% end %>
+</table>
+<div class="text-center p-2">
+  <button class="btn btn-info"><%= link_to 'Home', authors_path(@author), :class => "text-decoration-none text-dark" %></Button>
+  <button class="btn btn-warning"><%= link_to 'Edit', edit_author_path(@author), :class => "text-decoration-none text-dark" %></Button>
+  <button class="btn btn-danger"><%= link_to "Destroy", author_path(@author), data: { turbo_method: :delete, turbo_confirm: "Are you sure?" }, :class => "text-decoration-none text-light" %></Button>
```
Comment (poonamsimform) [inline app/views/authors/show.html.erb]: Use new syntax: `class: "text-decoration-none text-dark"`

File: db/migrate/20230307085416_add_initial_books.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class AddInitialBooks < ActiveRecord::Migration[7.0]
+  def up
+    count = Author.all.count
+    8.times do |i|
+      Book.create(name: "Book ##{i+1}", price: rand(100..1000),author_id: rand(1..count))
```
Comment (poonamsimform) [inline db/migrate/20230307085416_add_initial_books.rb]: space after comma.

File: db/migrate/20230307090837_add_initial_authors.rb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+class AddInitialAuthors < ActiveRecord::Migration[7.0]
+  def up
+    author_firstname = ["Dhyey", "Krutik", "Aditya","William","J.K."]
+    author_lastname = ["Sapara", "Patel", "Kansara","Shakesphere","Rowling"]
+    author_dob = ["15-12-2001","6-4-1992","22-09-1980","23-04-1994","23-06-1965"]
+    author_email = ["dhyeysapara@gmail.com", "krutik.patel2@hotmail.com", "aditya_kansara4@yahoo.com","williamsphere@hotmail.com","rowling62@gmail.com"]
```
Comment (poonamsimform) [inline db/migrate/20230307090837_add_initial_authors.rb]: Space after comma

File: db/migrate/20230307095453_fix_books_price.rb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+class FixBooksPrice < ActiveRecord::Migration[7.0]
+  def change
+    reversible do |rev|
+      change_table :books do |t|
+        rev.up { t.change :price, :decimal, precision: 6, scale: 1}
```
Comment (poonamsimform) [inline db/migrate/20230307095453_fix_books_price.rb]: Space before `}` braces

File: db/seeds.rb [hunk 9]
```diff
@@ -5,3 +5,8 @@
 #
 #   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
 #   Character.create(name: "Luke", movie: movies.first)
+
+10.times do |i|
+  count = Author.all.count
+  Book.find_or_create_by(name: "Book ##{rand(1000)}", price: rand(100..1000),author_id: rand(1..count))
```
Comment (poonamsimform) [inline db/seeds.rb]: comma after space

**Comment by poonamsimform** [comment]
Please check all the files once and use the new syntax to define a class in the tag. Also, check indentation.

File: app/views/authors/show.html.erb [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+<table class="table w-50 table-bordered table-striped table-hover m-auto text-center">
+  <tr>
+    <th>Id</th>
+    <th>Name</th>
+    <th>Date Of Birth</th>
+    <th>Email</th>
+  </tr>
+  <tr>
+    <td><%= @author.id %></td>
+    <td><%= @author.first_name + " " + @author.last_name %></td>
```
Comment (poonamsimform) [inline app/views/authors/show.html.erb]: Please use interpolation here...

File: app/views/authors/show.html.erb [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+<table class="table w-50 table-bordered table-striped table-hover m-auto text-center">
+  <tr>
+    <th>Id</th>
+    <th>Name</th>
+    <th>Price</th>
+    <th>Author Id</th>
+  </tr>
+  <% @books.each do |book| %>
+    <tr>
+      <td><%= link_to book.id, book %></td>
```
Comment (poonamsimform) [inline app/views/authors/show.html.erb:29]: Any reason to add a `link` here?

Comment (Dhyey54) [reply app/views/authors/show.html.erb:29]: I added link here if anyone wants to check details about some specific book of the author.

**Review by poonamsimform — APPROVED** [review]

**Pull request merged**